Source/WTF/wtf/text/AtomStringImpl.h

```diff
@@ -130,9 +130,9 @@ ALWAYS_INLINE Ref<AtomStringImpl> AtomStringImpl::add(StringImpl& string)
 
 ALWAYS_INLINE Ref<AtomStringImpl> AtomStringImpl::add(Ref<StringImpl>&& string)
 {
-    if (string->isAtom()) {
+    if (is<AtomStringImpl>(string)) {
```

Comment:
Why not `dynamicDowncast`?

Reply:
There is not a way to write this code as efficiently with dynamicDowncast or without introducing a use-after-move:
```
if (RefPtr atomStringImpl = dynamicDowncast<AtomStringImpl>(string)) {
    return atomStringImpl;
return addSlowCase(WTFMove(string));
```
Slower because it causes an extra ref of the StringImpl.

```
if (RefPtr atomStringImpl = dynamicDowncast<AtomStringImpl>(WTFMove(string))) {
    return atomStringImpl;
return addSlowCase(WTFMove(string));
```
No extra ref but we're doing a use-after-move.